vividus-plugin-rest-api/src/main/java/org/vividus/bdd/steps/api/HttpRequestSteps.java

```diff
@@ -202,6 +204,39 @@ public void setCustomRequestConfig(ExamplesTable configItems) throws ReflectiveO
         httpTestContext.putRequestConfig(requestConfigBuilder.build());
     }
 
+    /**
+     * Step sends HTTP request for the given <b>url</b> and waits for a specified amount of time
+     * until HTTP response code is equal to what is expected and stores HTTP response into Scenario-level variable.<br>
+     * This step can use the request body that was set before.
+     * Request body shouldn't be set for methods that can't contain body (GET, HEAD, OPTIONS, TRACE),
+     * and should be set for methods that must contain body (PATCH, POST, PUT).
+     *
+     * @param responseCode for example 200, 404
+     * @param httpMethod   HTTP method type. Parameter accepts the following HTTP methods:
+     *                     <ul>
+     *                     <li>GET</li>
+     *                     <li>HEAD</li>
+     *                     <li>POST</li>
+     *                     <li>PUT</li>
+     *                     <li>OPTIONS</li>
+     *                     <li>DELETE</li>
+     *                     <li>TRACE</li>
+     *                     <li>PATCH</li>
+     *                     </ul>
+     * @param url          for example https://www.vividus.org/sitemap.xml
+     * @param duration     Time duration to wait
+     * @param retryTimes   How many times request will be retried; duration/retryTimes=timeout between requests
+     * @throws IOException If an input or output exception occurred
+     */
+    @When("When I wait for '$responseCode' response code for HTTP $httpMethod request for a resource "
+            + "with the URL '$url' for '$duration' duration retrying $retryTimes times")
```

Comment:
@When("I wait for `$responseCode` response code for HTTP $httpMethod request to resource "
           + "`$resourceUrl` for `$duration` duration retrying $retryTimes times")

Reply:
Also what about making this step a bit more universal:

```
@When("I wait for response code `$responseCode` for `$duration` duration retrying $retryTimes times"
        + "$stepsToExecute")
```
?